src/generic/grideditors.cpp

```diff
@@ -1710,5 +1711,107 @@ wxGridCellAutoWrapStringEditor::Create(wxWindow* parent,
                                     wxTE_MULTILINE | wxTE_RICH);
 }
 
+#if wxUSE_DATEPICKCTRL
+
+// ----------------------------------------------------------------------------
+// wxGridCellDateEditor
+// ----------------------------------------------------------------------------
+
+wxGridCellDateEditor::wxGridCellDateEditor()
+{
+}
+
+void wxGridCellDateEditor::Create(wxWindow* parent, wxWindowID id,
+                                  wxEvtHandler* evtHandler)
+{
+    m_control = new wxDatePickerCtrl(parent, id);
+
+    wxGridCellEditor::Create(parent, id, evtHandler);
+}
+
+void wxGridCellDateEditor::SetSize(const wxRect& r)
+{
+    wxRect rect = r;
+    wxSize bestSize = m_control->GetBestSize();
+    bool resize = false;
+
+    if( rect.GetHeight() < bestSize.GetHeight() )
+    {
+        rect.SetHeight(bestSize.GetHeight());
+        resize = true;
+    }
+    if( rect.GetWidth() < bestSize.GetWidth() )
+    {
+        rect.SetWidth(bestSize.GetWidth());
+        resize = true;
+    }
+
+    if ( resize )
+    {
+        m_control->SetSize(rect.GetSize());
+    }
+
+    wxGridCellEditor::SetSize(rect);
+}
+
+void wxGridCellDateEditor::BeginEdit(int row, int col, wxGrid* grid)
+{
+    wxASSERT_MSG(m_control, wxT("The wxGridCellDateEditor must be created first!"));
```

Comment:
Please don't use `wxT` in the new code. Either omit it completely (fine in 99% of the cases) or use `wxS()` in the really performance-critical code to avoid unnecessary conversions.